c++/src/Compression.cc

```diff
@@ -636,6 +884,33 @@ DIAGNOSTIC_POP
     return static_cast<uint64_t>(result);
   }
 
+  std::unique_ptr<BufferedOutputStream>
+     createCompressor(
+                      CompressionKind kind,
+                      OutputStream * outStream,
+                      CompressionStrategy strategy,
+                      uint64_t bufferCapacity,
+                      uint64_t blockSize,
+                      MemoryPool& pool) {
+    switch (static_cast<int64_t>(kind)) {
+    case CompressionKind_NONE: {
+      return std::unique_ptr<BufferedOutputStream>
+        (new BufferedOutputStream(pool, outStream, bufferCapacity, blockSize));
+    }
+    case CompressionKind_ZLIB: {
+      int level = strategy == CompressionStrategy_SPEED ? 1 : 9;
```

Comment:
We should keep the default level to `Z_DEFAULT_COMPRESSION` (-1). Only if `strategy` is specified, we must set accordingly. I think the Java code does something similar.